modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/QueryMetadataIntegrationTest.java

```diff
@@ -0,0 +1,344 @@
+
+                assertEquals(expMeta.schemaName(), actualMeta.schemaName());
+                assertEquals(expMeta.typeName(), actualMeta.typeName());
+                assertEquals(expMeta.fieldName(), actualMeta.fieldName());
+                assertEquals(expMeta.fieldTypeName(), actualMeta.fieldTypeName());
+                assertEquals(expMeta.nullability(), actualMeta.nullability());
+                assertEquals(expMeta.scale(), actualMeta.scale());
+                assertEquals(expMeta.precision(), actualMeta.precision());
+            }
+        }
+    }
+
+    /** */
+    private static class MetadataBuilder {
+        /** */
+        private final List<GridQueryFieldMetadata> fields = new ArrayList<>();
+
+        /** */
+        private final int fieldOffset;
+
+        /** */
+        MetadataBuilder() {
+            this(0);
+        }
+
+        /** */
+        MetadataBuilder(int fieldOffset) {
+            this.fieldOffset = fieldOffset;
+        }
+
+        /** */
+        MetadataBuilder add(Class<?> fieldClass) {
+            add(fieldClass, PRECISION_NOT_SPECIFIED, SCALE_NOT_SPECIFIED);
+            return this;
+        }
+
+        /** */
+        MetadataBuilder add(Class<?> fieldClass, int precision, int scale) {
+            add(null, null, fieldClass, "?" + (fields.size() + fieldOffset), precision, scale, true);
+            return this;
+        }
+
+        /** */
+        MetadataBuilder add(String schemaName, String typeName, Class<?> fieldClass, String fieldName) {
+            add(schemaName, typeName, fieldClass, fieldName, true);
+            return this;
+        }
+
+        /** */
+        MetadataBuilder add(
+            String schemaName,
+            String typeName,
+            Class<?> fieldClass,
+            String fieldName,
+            boolean isNullable
+        ) {
+            add(schemaName, typeName, fieldClass, fieldName, PRECISION_NOT_SPECIFIED, SCALE_NOT_SPECIFIED, isNullable);
+            return this;
+        }
+
+        /** */
+        MetadataBuilder add(
+            String schemaName,
+            String typeName,
+            Class<?> fieldClass,
+            String fieldName,
+            int precision,
+            int scale,
+            boolean isNullable
+        ) {
+            GridQueryFieldMetadata mocked = Mockito.mock(GridQueryFieldMetadata.class);
+
+            Mockito.when(mocked.schemaName()).thenReturn(schemaName);
+            Mockito.when(mocked.typeName()).thenReturn(typeName);
+            Mockito.when(mocked.fieldName()).thenReturn(fieldName);
+            Mockito.when(mocked.fieldTypeName()).thenReturn(fieldClass.getName());
+            Mockito.when(mocked.precision()).thenReturn(precision);
+            Mockito.when(mocked.scale()).thenReturn(scale);
+            Mockito.when(mocked.nullability())
+                .thenReturn(isNullable ? ResultSetMetaData.columnNullable : ResultSetMetaData.columnNoNulls);
```

Comment:
CalciteQueryFieldMetadata - is a simple POJO and looks like it can be used here instead of mocking. This will simplify the code and prevent adding the new dependency.

Reply:
I disagree here -- mocking is the best option here.